src/extension/chatSessions/vscode-node/copilotCLIChatSessionsContribution.ts

```diff
@@ -154,36 +155,50 @@ export class CopilotCLIChatSessionItemProvider extends Disposable implements vsc
 
 	public async provideChatSessionItems(token: vscode.CancellationToken): Promise<vscode.ChatSessionItem[]> {
 		const sessions = await this.copilotcliSessionService.getAllSessions(token);
-		const diskSessions = sessions.map(session => this._toChatSessionItem(session));
+		const diskSessions: vscode.ChatSessionItem[] = [];
+		for (const session of sessions) {
```

Comment:
Promise.all(... would be be, as we'd be processing & awaiting on all promises in parallel as opposed to processing each session one at at time sequentially.